From: Benjamin Drung <benjamin.drung@canonical.com>
Date: Tue, 1 Sep 2026 19:59:04 +0200
Subject: fix(dmsquash-live): support umount applet from busybox

busybox's `umount` applet does not support the long form `--detach-loop`
but it supports the short form `-d`.

Fixes: https://github.com/dracut-ng/dracut/issues/2684
Part of: https://github.com/dracut-ng/dracut/issues/2437
Origin: upstream, https://github.com/dracut-ng/dracut/commit/8bc69aeda0c676889e9fd593547a5695b5fae603
---
 modules.d/70dmsquash-live/apply-live-updates.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/modules.d/70dmsquash-live/apply-live-updates.sh b/modules.d/70dmsquash-live/apply-live-updates.sh
index be3b2c2..febad45 100755
--- a/modules.d/70dmsquash-live/apply-live-updates.sh
+++ b/modules.d/70dmsquash-live/apply-live-updates.sh
@@ -17,6 +17,6 @@ fi
 # release resources on iso-scan boots with rd.live.ram
 if [ -d /run/initramfs/isoscan ] \
     && [ -f /run/initramfs/squashed.img ] || [ -f /run/initramfs/rootfs.img ]; then
-    umount --detach-loop /run/initramfs/live
+    umount -d /run/initramfs/live
     umount /run/initramfs/isoscan
 fi
